src/client/sandbox/event/simulator.js

```diff
@@ -697,49 +741,47 @@ export default class EventSimulator {
     contextmenu (el, options) {
         return this._simulateEvent(el, 'contextmenu', options, {
             button:  eventUtils.BUTTON.right,
-            which:   eventUtils.WHICH_PARAMETER.rightButton,
-            buttons: eventUtils.BUTTONS_PARAMETER.rightButton
+            buttons: eventUtils.BUTTONS_PARAMETER.noButton
         });
     }
 
     mousedown (el, options = {}) {
-        options.button  = options.button === void 0 ? eventUtils.BUTTON.left : options.button;
-        options.which   = options.which === void 0 || options.button !== eventUtils.BUTTON.right
-            ? eventUtils.WHICH_PARAMETER.leftButton
-            : eventUtils.WHICH_PARAMETER.rightButton;
-        options.buttons = options.buttons === void 0 ? eventUtils.BUTTONS_PARAMETER.leftButton : options.buttons;
+        const button  = options.button === void 0 ? eventUtils.BUTTON.left : options.button;
+        const buttons = button === eventUtils.BUTTON.left ? eventUtils.BUTTONS_PARAMETER.leftButton : eventUtils.BUTTONS_PARAMETER.rightButton;
+
+        options.button  = button;
+        options.buttons = options.buttons === void 0 ? buttons : options.buttons;
 
         return this._simulateEvent(el, 'mousedown', options);
     }
 
     mouseup (el, options = {}) {
-        options.button  = options.button === void 0 ? eventUtils.BUTTON.left : options.button;
-        options.which   = options.which === void 0 || options.button !== eventUtils.BUTTON.right
-            ? eventUtils.WHICH_PARAMETER.leftButton
-            : eventUtils.WHICH_PARAMETER.rightButton;
-        options.buttons = options.buttons === void 0 ? eventUtils.BUTTONS_PARAMETER.leftButton : options.buttons;
+        const button  = options.button === void 0 ? eventUtils.BUTTON.left : options.button;
 
-        return this._simulateEvent(el, 'mouseup', options);
+        return this._simulateEvent(el, 'mouseup', options, {
+            button,
+            buttons: eventUtils.BUTTONS_PARAMETER.noButton
+        });
     }
 
     mouseover (el, options) {
-        return this._simulateEvent(el, 'mouseover', options);
+        return this._simulateEvent(el, 'mouseover', EventSimulator._updateMouseEventButtonProperties(options));
```

Comment:
It's difficult to read. Move to separate variable.
Also, `EventSimulator._updateMouseEventButtonProperties(options)` call is not clean.
The function calculates `button`, `buttons` and `which` option values. And it's an implementation detail.
I propose to rename it as `_prepareMouseEventOptions (option)`.
So the final call will look like:
```js
mouseover (el, options) {
  const options = EventSimulator._prepareMouseEventOptions(options);
 
  return return this._simulateEvent(el, 'mouseover', options);
}
```